tools/serverpod_cli/lib/src/analyzer/entities/stateful_analyzer.dart

```diff
@@ -2,39 +2,43 @@ import 'package:serverpod_cli/analyzer.dart';
 import 'package:serverpod_cli/src/generator/code_generation_collector.dart';
 import 'package:serverpod_cli/src/util/protocol_helper.dart';
 
+class _ProtocolState {
+  ProtocolSource source;
+  SerializableEntityDefinition? entity;
+
+  _ProtocolState({
+    required this.source,
+  });
+}
+
 class StatefulAnalyzer {
-  final Map<String, SerializableEntityAnalyzer> _analyzers = {};
+  final Map<String, _ProtocolState> _protocolStates = {};
   List<SerializableEntityDefinition> _entities = [];
 
   Function(Uri, CodeGenerationCollector)? _onErrorsChangedNotifier;
 
-  /// Validates all protocols by running the validator twice to make sure all
-  /// references are resolved. The state is preserved to make future validations
-  /// less expensive. This method is required to use for the initialization of
-  /// the state. Subsequent validations should use [validateAll] or [validateProtocol].
+  /// Loads all yaml protocols and initializes the state. The state is preserved
+  /// to make future validations less expensive.
+  /// Subsequent validations should use [validateAll] or [validateProtocol].
   List<SerializableEntityDefinition> initialValidation(
```

Comment:
Question: Would it be possible to add this to the construction of the class?

I think this might be easier to addressed in https://github.com/serverpod/serverpod/pull/1098, but if this method call Is required before the class is usable I would suggest calling it directly in the constructor. This removes the need for a user of this interface to understand the relationship between the methods before using the class. (The comments are helpful but removing the implicit dependency is even better)

Reply:
This method is removed in #1098 so this issue is resolved there.